client/commands.c

```diff
@@ -48,73 +50,221 @@ void processXferCommand(char *nextParam, FILE *out, int ispwd, uint32_t* pwd);
+    { "keyGen"     , xdag_com_keyGen     , "generate new private/public key pair and set it by default" },
+    { "level"      , xdag_com_level      , "[N] print level of logging or set it to N (0 - nothing, ..., 9 - all)" },
+    { "miners"     , xdag_com_miners     , "for pool, print list of recent connected miners" },
+    { "mining"     , xdag_com_mining     , "[N] print number of mining threads or set it to N" },
+    { "net"        , xdag_com_net        , "run transport layer command, try 'net help'" },
+    { "pool"       , xdag_com_pool       , "[miners:maxip:maxconn:fee:reward:direct:fund]" },
+    { "run"        , xdag_com_run        , "run node after loading local blocks if option -r is used" },
+    { "state"      , xdag_com_state      , "print the program state" },
+    { "stats"      , xdag_com_stats      , "print statistics for loaded and all known blocks" },
+    { "terminate"  , xdag_com_terminate  , "terminate both daemon and this program" },
+    { "xfer"       ,(xdag_com_func_t)NULL, "[A] xfer S A, transfer S our %s to the address A" },
+    { "help"       , xdag_com_help       , "print xdag commands help" },
+    { (char *)NULL ,(xdag_com_func_t)NULL, (char *)NULL }
+};
+
+int xdag_com_account(char* args, FILE* out) {
+    processAccountCommand(args, out);
+    return 0;
+}
+
+int xdag_com_balance(char * args, FILE* out) {
+    processBalanceCommand(args, out);
+    return 0;
+}
+
+int xdag_com_block(char * args, FILE* out) {
+    processBlockCommand(args, out);
+    return 0;
+}
+
+int xdag_com_lastblocks(char * args, FILE* out) {
+    processLastBlocksCommand(args, out);
+    return 0;
+}
+
+int xdag_com_keyGen(char * args, FILE* out) {
+    processKeyGenCommand(out);
+    return 0;
+}
+
+int xdag_com_level(char * args, FILE* out) {
+    processLevelCommand(args, out);
+    return 0;
+}
+
+int xdag_com_mining(char * args, FILE* out) {
+    processMiningCommand(args, out);
+    return 0;
+}
+
+int xdag_com_net(char * args, FILE* out) {
+    processNetCommand(args, out);
+    return 0;
+}
+
+int xdag_com_pool(char * args, FILE* out) {
+    processPoolCommand(args, out);
+    return 0;
+}
+
+int xdag_com_miners(char * args, FILE* out) {
+    processMinersCommand(args, out);
+    return 0;
+}
+
+int xdag_com_stats(char * args, FILE* out) {
+    processStatsCommand(out);
+    return 0;
+}
+
+int xdag_com_state(char * args, FILE* out) {
+    fprintf(out, "%s\n", get_state());
+    return 0;
+}
+
+int xdag_com_run(char * args, FILE* out) {
+    g_xdag_run = 1;
+    return 0;
+}
+
+int xdag_com_terminate(char * args, FILE* out) {
+    processExitCommand();
+    return -1;
+}
+
+int xdag_com_exit(char * args, FILE* out) {
+    processExitCommand();
+    return -1;
+}
+
+int xdag_com_help(char *args, FILE* out) {
+    printHelp(out);
+    return 0;
+}
+
+char ** xdag_com_completion(const char *text, int start, int end) {
+    char **matches;
+    matches = (char **)NULL;
```

Comment:
Feel free to combine declaration and assign `char **matches = (char **)NULL;`

Reply:
ok